Fixes #29990 - stop clear API response on polling

Ron-Lavi opened the pull request:
Each polling response is getting reset on every new request
since initially in the APIReducer we set response to null.

Thanks @MariaAga for bringing that up :)


<!---

Thank you for contributing to The Foreman, please read the
[following guide](https://www.theforeman.org/contribute.html), in short:

* [Create an issue](https://projects.theforeman.org/projects/foreman/issues)
* Reference the issue via `Fixes #1234` in the commit message
* Prefer present-tense, imperative-style commit messages
* Mark all strings for translation, see [1]
* Suggest prerequisites for testing and testing scenarios following example above.
* Prepend `[WIP]` for work in progress to prevent bots from triggering actions
* Be patient, we will do our best to take a look as soon as we can
* Explain the purpose of the PR, attach screenshots if applicable
* List all prerequisites for testing (e.g. VMware cluster, two smart proxies...)
* Reviewers often use extensive list of items to check, have a look prior submitting [2]
* Be nice and respectful

1: https://projects.theforeman.org/projects/foreman/wiki/Translating#Translating-for-developers
2: https://github.com/theforeman/foreman/blob/develop/developer_docs/pr_review.asciidoc
-->

Ron-Lavi commented on webpack/assets/javascripts/react_app/redux/API/APIReducer.js:
```diff
@@ -14,8 +14,9 @@ const apiReducer = (state = initialState, { type, key, payload, response }) => {
         [key]: {
-          payload,
           response: null,
+          ...state[key],
```
For single API request there won't be any difference since the response is initially set to null as before,
though if the response already exist it will override the null value.

MariaAga reviewed (dismissed):
Thanks for this fix!

ekohl reviewed (commented):
Git commits should describe the solution in the title, not the problem. This description suggests you're resetting every request for polling while the code suggests you're doing the opposite.

ronlavi2412 committed:
Fixes #29990 - stop clear API response on polling

Each polling response is getting reset on every new request
since initially in the APIReducer we set response to null.

Ron-Lavi commented:
Thanks @ekohl, makes sense.. changed the commit message into
`Fixes #29990 - stop clear API response on polling` with the same description as before

ekohl reviewed (commented):
Just commenting no the description - that makes more sense now. Not going to comment on the actual code so no more objections from my side.

ezr-ondrej reviewed (approved):
Thanks @laviro :+1:
